app/views/hosts/_form.html.erb

```diff
@@ -138,6 +138,6 @@
 
     <%= hidden_field_tag 'bare_metal_capabilities', @host.bare_metal_capabilities, :disabled => true %>
     <%= f.hidden_field :overwrite? %>
-    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => resource_prev_url_with_search_filters || (@host.new_record? ? hosts_path : host_path(:id => @host.name_was)) %>
+    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => resource_prev_url_with_search_filters || (@host.new_record? ? hosts_path : current_host_details_path(@host)) %>
```

Comment:
Oh, this will not work in some cases:
1. change hostname
2. submit, but have some validation failure
3. cancel

Reply:
This was introduced as a fix for https://projects.theforeman.org/issues/4381

Reply:
Tho testing it it doesn't feel broken